src/lib.rs

```diff
@@ -193,9 +193,28 @@ impl<T: Float> Complex<T> {
     /// Computes `e^(self)`, where `e` is the base of the natural logarithm.
     #[inline]
     pub fn exp(self) -> Self {
-        // formula: e^(a + bi) = e^a (cos(b) + i*sin(b))
-        // = from_polar(e^a, b)
-        Self::from_polar(self.re.exp(), self.im)
+        // formula: e^(a + bi) = e^a (cos(b) + i*sin(b)) = from_polar(e^a, b)
+
+        // Treat the corner cases +∞, -∞, and NaN
+        let mut im = self.im;
+        if self.re.is_infinite() {
+            if self.re < T::zero() {
+                if !self.im.is_finite() {
+                    im = T::one();
```

Comment:
Where does this 1.0 come from -- is that an arbitrary choice?

I think the result will be 0+0i, so why not just return that?

Reply:
It's indeed an arbitrary choice (following the clang C++ implementation). I agree that it would be simpler to just return 0+0i, so I just implemented, tested, and git-pushed that.